state-sync/storage-service/server/src/lib.rs

```diff
@@ -705,11 +704,18 @@ fn log_storage_response(storage_response: &Result<StorageServiceResponse, Storag
                 sample!(
                     SampleRate::Duration(Duration::from_secs(SUMMARY_LOG_FREQUENCY_SECS)),
                     {
+                        let response = format!("{}", storage_response); // Use display formatting
                         debug!(LogSchema::new(LogEntry::SentStorageResponse).response(&response));
                     }
                 );
             } else {
-                debug!(LogSchema::new(LogEntry::SentStorageResponse).response(&response));
+                sample!(
+                    SampleRate::Duration(Duration::from_secs(SUMMARY_LOG_FREQUENCY_SECS)),
+                    {
+                        let response = format!("{}", storage_response); // Use display formatting
```

Comment:
we can probably remove the "// use display formatting", I don't think it says anything useful. Maybe it intended to imply that display > debug but ...

Reply:
why not put the sample! further up or even do something like ...

```
fn log_storage_response() {
  sample!(log_storage_response_sampled_internal())
}

Reply:
I think the point is to not sample the logs in case of error.

Reply:
good call, too much nesting there ;)